channeldb/graph.go

```diff
@@ -2143,29 +2174,16 @@ func (c *ChannelGraph) FilterKnownChanIDs(chansInfo []ChannelUpdateInfo,
 					zombieIndex, scid,
 				)
 
-				isStillZombie := isZombieChan(
-					info.Node1UpdateTimestamp,
-					info.Node2UpdateTimestamp,
-				)
-
-				switch {
-				// If the edge is a known zombie and if we
-				// would still consider it a zombie given the
-				// latest update timestamps, then we skip this
-				// channel.
-				case isZombie && isStillZombie:
+				// NOTE: In the past we used to resurrect zombie
+				// channels based on the timestamp value
+				// included in the `ReplyChannelRange` msg,
+				// however this is too risky because there is no
+				// proof that the peer has an `ChanUpdate` msg
+				// with this timestamp or could even just lie to
+				// us. Therefore to resurrect zombie channel we
+				// have to receive the actual `ChanUpdate` msg.
+				if isZombie {
```

Comment:
See https://github.com/lightningnetwork/lnd/pull/8030/files#r1394730858.

I flagged this potential issue on the previous PR, and @ellemouton explained that we need zombies to be marked live before the channel announcement comes.  Since we're reverting that change, we should make sure channel announcements are still handled correctly.

Reply:
Hmm thinking about it twice and reading through the thread of the prior issue, I think we can somehow rely on the timestamp in resurrecting the zombie channels, however we should probably built in some banning protection if the peer promises a ChanUpdate with a specific timestamp but is not sending it.

Reply:
The query msges, at least the range queries built on trust anyways so having a proper banning system in place is more important rather than not trusting the data in the first place ?

Reply:
Just want to point out that resurrecting a zombie channel just means removing it from the index and _not_ inserting it into the `edgeIndexBucket`. So I think we can keep it?